crates/neon/src/types_impl/extract/json.rs

```diff
@@ -73,17 +91,15 @@ impl<'cx, T> TryFromJs<'cx> for Json<T>
 where
     for<'de> T: serde::de::Deserialize<'de>,
 {
-    type Error = serde_json::Error;
```

Comment:
This was a mistake. It tightly couples the implementation to `serde_json`. In the future, if Node-API gets optimized object construction, we change the implementation to direct transcoding. Hiding away the error details gives us the flexibility.